homeassistant/components/risco/binary_sensor.py

```diff
@@ -150,3 +149,39 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
 
     async def _bypass(self, bypass: bool) -> None:
         await self._zone.bypass(bypass)
+
+
+class RiscoLocalAlarmedBinarySensor(BinarySensorEntity):
+    """Representation whether a zone in Risco local is currently triggering an alarm."""
```

Comment:
This is what an `alarm_control_panel` is for? It supports the `triggered` state.

Reply:
This is not for the whole alarm, but for a specific zone (e.g. motion sensor). This is the only way to figure out which zone triggered the alarm.
In other words, when the alarm is triggered, both the `alaram_control_panel` and the relevant zones, change their state to `triggered`

Reply:
(initially it was an attribute on the zone, but I was asked to separate it into a new entity)

Reply:
A Risco system is comprised of partitions and zones. Partitions are actual alarms - they can be armed, disarmed, triggered, and are represented by alarm control panels. Zones, on the other hand, are detectors (e.g. a window sensor, a motion sensor in the kitchen). They cannot be armed or disarmed individually. 
Zones are already represented as binary sensors that reflect when they’re detected. This PR adds an additional binary sensor per zone (each zone also has a device) to signal whether this specific zone triggered an alarm. It’s a single binary value, that’s not directly related to the state of the alarm. Initially I had this as an attribute on the main binary sensor for the zone, but was asked to split it into a separate one. 
Hope this clears it up.

Reply:
@frenck does this address your question?

Reply:
While we're waiting, some additional context:
The proposed binary sensor acts nothing like an alarm - it acts like a binary sensor with two states. The only way in which it's somewhat close to an alarm control panel, is that its `on` state can be interpreted in the real world as "triggering", which is close to `triggered`. However:
1. Its `off` state does not map to any alarm control panel state
2. With the exception, perhaps, of `triggered`, none of the alarm control panel states make any sense here
3. None of the alarm control panel services are relevant

Reply:
If the zone can only have two states, triggered and off, it makes sense to make it a binary sensor and not an alarm control panel entity.

Reply:
To be fair, the zones itself can have multiple states - it's a collection of several binary sensors that are each independent:
* Not detected / detected (this is the main binary sensor entity for the zone)
* Triggering an alarm / not triggering an alarm (this binary sensor)
* Bypassed / not bypassed (currently an attribute, will be split into a binary sensor in a future PR)
* Armed / disarmed

As far as I'm aware, bypassed zones can not be considered armed, but all of the other combinations are possible (for example, certain zones can trigger an alarm even if they're not armed, a zone is still considered triggering even when it's no longer armed, etc...)
Because of that independence, I still think a collection of binary sensors is the right abstraction, even once all of these are implemented, despite some of these states resembling an alarm control panel.

Reply:
I think the three last cases (triggered/bypassed/armed/disarmed) could be combined into one alarm control panel entity. A triggered state takes preference over the other states.

Reply:
How would you represent bypassed as an alarm control panel? And which of the arming states would you use for armed?

Regardless, I think it's a confusing abstraction for Risco owners, who don't think of their zones as individual alarms, but rather as detectors.

Reply:
Yeah, it seems better to use binary sensors. We'd shoehorn the states into the alarm panel state definition otherwise.